engine/src/main/java/io/zeebe/engine/processor/RecordValueCache.java

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Zeebe Community License 1.0. You may not use this file
+ * except in compliance with the Zeebe Community License 1.0.
+ */
+package io.zeebe.engine.processor;
+
+import static io.zeebe.engine.processor.TypedEventRegistry.EVENT_REGISTRY;
+
+import io.zeebe.logstreams.log.LoggedEvent;
+import io.zeebe.protocol.impl.record.UnifiedRecordValue;
+import io.zeebe.protocol.record.ValueType;
+import io.zeebe.util.ReflectUtil;
+import java.util.Collections;
+import java.util.EnumMap;
+import java.util.Map;
+
+public class RecordValueCache {
```

Comment:
I think the name is a bit confusing because it is not really a cache. What about `RecordValueFactory`?

Reply:
=> RecordValues